### ISSUE TITLE
An expression tree may not contain an assignment operator

### ISSUE THREAD

**Issue by johnnymo87** [BaylorRae/Let.cs#1]
I like the library! I'm new to C#, and I miss some of the sugar of ruby libraries, especially rspec's `let`.

I hand-rolled my own version of [NBuilder](https://github.com/nbuilder/nbuilder), and was saddened to see that it made things incompatible with this library.
```cs
private Client client => Let(() => {
    return clientSeeder.Build().With(
        c => c.ClientName = "Main Client"
    ).Persist();
});
```
`An expression tree may not contain an assignment operator`

This isn't an issue, I just wanted to leave it here for the record.

**Comment by BaylorRae** [comment]
Thanks for the comment, I hadn't seen NBuilder before and will definitely try it out when I have a chance! I've also run into the assignment error. My solution has been moving the expression into a method and defining my object there but I'm wondering if there's a better way. I'll have to play with it and see.

**Comment by BaylorRae** [comment]
I think I've found a solution to this problem. I've added an overload that accepts a string as the first parameter and a `Func<T>` as the second. This has been released in [v1.1.0](https://github.com/BaylorRae/Let.cs/releases/tag/v1.1.0).

```csharp
private Client client => Let("client", () => {
    return clientSeeder.Build().With(
        c => c.ClientName = "Main Client"
    ).Persist();
});
```
